bpatch_json.md

```diff
@@ -0,0 +1,160 @@
+  * Installation folder
+    * `bpatch`
+    * bpatch_a\\ACTIONS files
+    * bpatch_b\\binary data files from ACTIONS file (if files are needed)
+
+**NOTE:** It's possible to modify these folders using `-fa` for actions folder (AFN) and `-fb` for binary files folder (BFFN)
+
+## Actions File Sample
+An Actions file sample written in proven ASCII symbols and [`JSON`][JSON] format can be as follows:
+
+<details><summary><B>Sample with ACTIONS file (press triangle to expand)</B></summary>
+
+```json
+{
+    "dictionary":
+    {
+        "decimal":
+        {
+            "pattern name": [ 13, 10 ],
+            "another binary": [ 13 ]
+        },
+        "hexadecimal":
+        {
+            "hex pattern name": [ "0A", "09", "03" ],
+            "hex02": [ "00", "Fe", "3A" ]
+        },
+        "text":
+        {
+            "text name": "textual value",
+            "Hi": "Hello World!"
+        },
+        "file":
+        {
+            "fileX": "filename.bin"
+        },
+        "composite":
+        [
+            {
+                "composite 001":
+                [
+                    "pattern name",
+                    "text name",
+                    "hex pattern name"
+                ]
+            },
+            {
+                "complexOne":
+                [
+                    "hex02",
+                    "fileX",
+                    "Hi",
+                    "composite 001"
+                ]
+            }
+        ]
+    },
+    "todo":
+    [
+        {
+            "replace":
+            {
+                "pattern name": "hex pattern name",
+                "hex02": "complexOne"
+            }
+        },
+        {
+            "replace":
+            {
+                "text name": "fileX"
+            }
+        }
+    ]
+}
+```
+</details>
+
+**Note:** `bpatch` has own [`JSON`][JSON] parser and therefore
+  * Control characters in the text **cannot** be unicode. /uXXXX sequences will remain the same (ignored for transformation)
+  * Control [`JSON`][JSON] character sequences in the [`JSON`][JSON] text literals like \\b \\f \\n \\r \\t \\\\ \\/ \\" will be transformed and processed. For example text [`JSON`][JSON] object { "text\\"" : "meaning" } will have a name `text"` and value `meaning`
+  * [`JSON`][JSON] rules not allow line wrapping however for `bpatch` parser it is not a problem. Nonetheless, the recommendation is to use \\n or \\r instead of this trick
+
+
+## JSON Composition Rules
+1. ACTIONS file must contain root [`JSON`][JSON] object: { }
+1. root [`JSON`][JSON] object must contain `dictionary` and `todo` objects inside
+
+### dictionary
+ '`dictionary`' is defining set of lexemes to search in the SOURCE file and to write into DEST file
```

Comment:
Hmmmm, can't figure out the purpose of dictionary. The explanation from my point of view if not transparent

Reply:
After a conversation with you I suggest you to mention here the following dedails
1) I injected the names of lexemes to support composite lexemes and that's it.
2) A dictionary is a mapping from a lexeme name to its value, with the exception of composite types. You can read about how composite types work below.